zetaclient/chains/evm/signer/gas_test.go

```diff
@@ -0,0 +1,121 @@
+package signer
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/assert"
+	"github.com/zeta-chain/zetacore/x/crosschain/types"
+)
+
+func Test_makeGasFromCCTX(t *testing.T) {
+	logger := zerolog.Nop()
+
+	cctx1 := getCCTX(t)
+	cctx1.GetOutboundParams()[0].GasLimit = MinGasLimit - 200
+	cctx1.GetOutboundParams()[0].GasPrice = gwei(2).String()
+
+	cctx2 := getCCTX(t)
+	cctx2.GetOutboundParams()[0].GasLimit = 200_000
+	cctx2.GetOutboundParams()[0].GasPrice = gwei(3).String()
+
+	cctx3 := getCCTX(t)
+	cctx3.GetOutboundParams()[0].GasLimit = 2_000_000
+	cctx3.GetOutboundParams()[0].GasPrice = gwei(3).String()
+
+	for _, tt := range []struct {
```

Comment:
Some other cases would be helpful, such as cases where priorityFee is null, negative, etc.